api/v3/config.go

```diff
@@ -40,35 +40,53 @@ type ConfigEntry struct {
 	// SecretRef is used for sensitive configuration entries
 	// Mutually exclusive with Value
 	// +optional
-	SecretRef *SecretReference `json:"secretRef,omitempty"`
+	SecretRef *Reference `json:"secretRef,omitempty"`
+
+	// ConfigRef is used for configuration entries
+	// Mutually exclusive with Value
+	// +optional
+	ConfigRef *Reference `json:"configRef,omitempty"`
 }
 
-// SecretReference points to a value in a Kubernetes secret
-type SecretReference struct {
-	// Name is the name of the secret in the same namespace
+// Reference points to a value in a Kubernetes secret
+type Reference struct {
+	// Name is the name of the secret or configmap in the same namespace
 	// +required
 	Name string `json:"name"`
-	// Key is the key within the secret
+	// Key is the key within the secret or configmap
 	// +required
 	Key string `json:"key"`
 }
 
 // Validate ensures ConfigEntry has valid state
 func (c *ConfigEntry) Validate() error {
 	if c.Absent != nil && *c.Absent {
-		if c.Value != nil || c.SecretRef != nil {
-			return fmt.Errorf("absent entries cannot have value or secretRef")
+		if c.Value != nil || c.SecretRef != nil || c.ConfigRef != nil {
+			return fmt.Errorf("absent entries cannot have value, configRef or secretRef")
 		}
 		return nil
 	}
 
 	// For present entries, exactly one of Value or SecretRef must be set
 	hasValue := c.Value != nil
 	hasSecretRef := c.SecretRef != nil
+	hasConfigRef := c.ConfigRef != nil
 
-	if hasValue == hasSecretRef {
-		return fmt.Errorf("config entries can have either a value or a secretRef")
+	if !c.hasOnlyOneValueConfiguration(hasValue, hasSecretRef, hasConfigRef) {
+		return fmt.Errorf("config entries can have either a value, configRef or a secretRef")
 	}
 
 	return nil
 }
+
+func (c *ConfigEntry) hasOnlyOneValueConfiguration(bools ...bool) bool {
```

Comment:
```suggestion
	if !exclusiveOr(hasValue, hasSecretRef, hasConfigRef) {
		return fmt.Errorf("config entries can have either a value, configRef or a secretRef")
	}

	return nil
}

func exclusiveOr(bools ...bool) bool {
```
This does not need a method receiver and the more universal XOR describes this better.